File: src/main/java/tn/esprit/fundsphere/Entities/TransactionManagment/Transaction.java
package tn.esprit.fundsphere.Entities.TransactionManagment;


import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;
import tn.esprit.fundsphere.Entities.AccountManagment.Account;

import java.time.LocalDate;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@FieldDefaults(level = AccessLevel.PRIVATE)
@Table(name = "transaction")
@ToString
public class Transaction {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    long idTransaction;

    double amount;

    LocalDate date;

    @Enumerated(EnumType.STRING)
    TypeTransaction typeT;

    @ManyToOne
    @JoinColumn(name = "sender_id", nullable = false)
    private Account sender;

    @ManyToOne
    @JoinColumn(name = "receiver_id", nullable = false)
    private Account receiver;

}

File: angular/src/app/core/models/transaction.ts
import { Account } from "./account";
import { TypeTransaction } from "./type-transaction";

export class Transaction {
    idTransaction!: number;
    amount!: number;
    date!: Date;
    typeT!: string; // Assuming TypeTransaction is represented as a string in TypeScript
    sender!: Account; // Assuming Account model is defined
    receiver!: Account; // Assuming Account model is defined
    //loan?: any; // Assuming Loan model is defined
    //investment?: any; // Assuming Investment model is defined
}
